WorkflowSwiftUIExperimental/Sources/SwiftUIScreen.swift

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2023 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#if canImport(UIKit)
+
+import SwiftUI
+import Workflow
+import WorkflowUI
+
+public protocol SwiftUIScreen: Screen {
+    associatedtype Content: View
+
+    @ViewBuilder
+    static func makeView(model: ObservableValue<Self>) -> Content
```

Comment:
It's possible for this to be an instance method if we were directly [mapping Screen values to View values](https://docs.google.com/document/d/1DemHF9RViaNBUkp4T3rPqigGK2pZBJqtsyVVudncZCY/edit#heading=h.c3iosk6vfize). However, it's not clear how it can be an instance method when we want to [use `@ObservedObject`](https://docs.google.com/document/d/1DemHF9RViaNBUkp4T3rPqigGK2pZBJqtsyVVudncZCY/edit#heading=h.ofc332szrbkb). See the rest of that doc for why we've favored the latter so far.

However, I think we should take another pass at this soon. You're not the first to be repulsed by it, and some of the rationale may be stale in iOS 15.

Reply:
I took [another stab at conforming the rendering directly to View](https://github.com/square/workflow-swift/compare/tomb/swiftui-testbed-workflowswiftuiexperimental...tomb/swiftui-no-observableobject), eliminating the use of `ObservableObject`. It simplifies the implementation and the developer interface quite a bit.

In iOS 16 at least, I'm not seeing the animation issues (e.g. in Toggle) that we observed at the time of the design doc. This approach _might_ lead to more view body evaluations, but after my performance noodling last month, I think those evaluations are cheap as long as they don't trigger additional work in the Core Animation commit phase.

I'll dig more into this next week! cc @n8chur @watt

Reply:
Opening a draft PR from that branch: https://github.com/square/workflow-swift/pull/253